src/cacheout/cache.py

```diff
@@ -442,6 +442,29 @@ def expire_times(self) -> t.Dict[t.Hashable, T_TTL]:
         """
         with self._lock:
             return self._expire_times.copy()
+        
+    def item_ttl(self, key: t.Hashable) -> T_TTL:
+        """
+        Returns the remaining time to live of a key that has a TTL.
+
+        Args:
+            key: Cache key to review.
+
+        Returns:
+            float: the remaining time to live of item
+            int: ``-1`` if the key exists but has no associated expire, ``-2`` if key didn't exist.
+        """
+        with self._lock:
+            expire_time = self._expire_times.copy().get(key)
+            if not expire_time:
+                if not self._get(key):
+                    return -2
+                return -1
+            ttl = expire_time - self.timer()
+            if ttl > 0:
+                return ttl
+            self._delete(key)
+            return -2
```

Comment:
Would prefer to return `None` instead of `-1` or `-2`. Either the key has a remaining TTL or it doesn't (either because it was previously set and expired or it isn't set).

If needing to distinguish between `-1` or `-2` states then user could do `cache.has(key) and key not in cache.expire_items() -> -1` and `not cache.has(key) -> -2`. Just prefer not to overload the return values for this case and keep the return simpler. But I could see some other method to make getting the `-1` state easier (e.g. `cache.has_ttl(key)`).